src/storage/table_test.cc

```diff
@@ -504,26 +741,51 @@ TEST_F(TableTest, TableIteratorCount) {
         count++;
         it->Next();
     }
-    ASSERT_EQ(44471, count);
+    if (storageMode == ::openmldb::common::StorageMode::kMemory) {
+        ASSERT_EQ(44471, count);
+    } else {
+        ASSERT_EQ(55551, count);
+    }
+
     delete it;
 
-    for (int i = 0; i < 200000; i++) {
-        TableIterator* cur_it = table->NewTraverseIterator(0);
-        std::string key = "pk" + std::to_string(i);
-        cur_it->Seek(key, 9528);
-        ASSERT_TRUE(cur_it->Valid());
-        delete cur_it;
+    TableIterator* cur_it = table->NewTraverseIterator(0);
+    if (storageMode == ::openmldb::common::StorageMode::kMemory) {
+        for (int i = 0; i < 200000; i++) {
+            std::string key = "pk" + std::to_string(i);
+            cur_it->Seek(key, 9528);
+            ASSERT_TRUE(cur_it->Valid());
+        }
+    } else {
+        for (int i = 0; i < 100000; i = i + 2) {
+            std::string key = "pk" + std::to_string(i);
+            cur_it->Seek(key, 9528);
+            ASSERT_TRUE(cur_it->Valid());
+        }
     }
+    
+    delete cur_it;
+    
+}
+
+TEST_F(TableTest, TableIteratorCountMem) {
+    TableIteratorCount(::openmldb::common::StorageMode::kMemory);
+}
+
+TEST_F(TableTest, TableIteratorCountDisk) {
+    TableIteratorCount(::openmldb::common::StorageMode::kHDD);
 }
 
-TEST_F(TableTest, TableIteratorTS) {
+
+void TableIteratorTS(::openmldb::common::StorageMode storageMode) {
     ::openmldb::api::TableMeta table_meta;
     table_meta.set_name("table1");
-    table_meta.set_tid(1);
-    table_meta.set_pid(0);
+    table_meta.set_tid(17);
+    table_meta.set_pid(1);
```

Comment:
why we need to change the tid and pid?

Reply:
For Disktable, we will use tid and pid to locate the file in the disk. For example, a table with tid 16 and pid 1 will be stored in /tmp/.../16_1/ . Thus every test uses a different location and we can delete all these files in one time after all the tests are finished.

Reply:
You can define `TearDown` and `Setup` inside `TableTest` class.

They will be called for every `test` under `TableTest`.

Then no need to use different `tid` and `pid` for each test.

Reply:
It's a reasonable way. But will we have the need to run tests in parallel?